cmd/syncthing/gui.go

```diff
@@ -1019,8 +1019,13 @@ func (s *apiService) getEvents(w http.ResponseWriter, r *http.Request, eventSub
 	qs := r.URL.Query()
 	sinceStr := qs.Get("since")
 	limitStr := qs.Get("limit")
+	timeoutStr := qs.Get("timeout")
 	since, _ := strconv.Atoi(sinceStr)
 	limit, _ := strconv.Atoi(limitStr)
+	timeoutSec, timeoutErr := strconv.Atoi(timeoutStr)
+	if timeoutErr != nil || timeoutSec < 0 { // A timeout of 0 is valid
```

Comment:
I'd simplify to just

```go
timeoutSec, _ := strconv.Atoi(timeoutStr)
if timeoutSec <= 0 {
    timeoutSec = DefaultEventTimeoutS
}
```

but I don't care strongly about it (apart from having a constant which is already mentioned).